test/Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tests/Infrastructure/HttpClientExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Precompilation
+{
+    public static class HttpClientExtensions
+    {
+        public static async Task<string> GetStringWithRetryAsync(
+            this HttpClient httpClient, 
+            string url,
+            ILogger logger)
+        {
+            var response = await RetryHelper.RetryRequest(() => httpClient.GetAsync(url), logger, retryCount: 5);
```

Comment:
Hopefully we never need more than 2 (where the first one always fails and the second one passes). The Retry needs back offs so that it's not spamming the server as it's being set up. I'll follow up with some fixes in Hosting.